core/src/Kokkos_View.hpp

```diff
@@ -19,13 +19,1023 @@
+  static constexpr Impl::integral_constant<size_t, base_t::rank_dynamic()>
+      rank_dynamic = {};
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  enum {Rank KOKKOS_DEPRECATED_WITH_COMMENT("Use rank instead.") = rank()};
 #endif
 
-#include <View/Kokkos_ViewLegacy.hpp>
+  KOKKOS_INLINE_FUNCTION constexpr array_layout layout() const {
+    return Impl::array_layout_from_mapping<array_layout, mdspan_type>(
+        base_t::mapping());
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_0() const { return stride(0); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_1() const { return stride(1); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_2() const { return stride(2); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_3() const { return stride(3); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_4() const { return stride(4); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_5() const { return stride(5); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_6() const { return stride(6); }
+  KOKKOS_INLINE_FUNCTION constexpr size_t stride_7() const { return stride(7); }
+
+  template <typename iType>
+  KOKKOS_INLINE_FUNCTION constexpr std::enable_if_t<std::is_integral_v<iType>,
+                                                    size_t>
+  stride(iType r) const {
+    // base class doesn't have constraint
+    return base_t::stride(r);
+  }
+
+  template <typename iType>
+  KOKKOS_INLINE_FUNCTION void stride(iType* const s) const {
+    if constexpr (rank() > 0) {
+      for (size_t r = 0; r < rank(); r++) s[r] = base_t::stride(r);
+      s[rank()] = s[rank() - 1] * base_t::extent(rank() - 1);
+    }
+  }
+
+  //----------------------------------------
+  // Range span is the span which contains all members.
+
+  enum {
+    reference_type_is_lvalue_reference =
+        std::is_lvalue_reference_v<reference_type>
+  };
+
+  KOKKOS_INLINE_FUNCTION constexpr size_t span() const {
+    return base_t::mapping().required_span_size();
+  }
+  KOKKOS_INLINE_FUNCTION bool span_is_contiguous() const {
+    return base_t::is_exhaustive();
+  }
+  KOKKOS_INLINE_FUNCTION constexpr bool is_allocated() const {
+    return data() != nullptr;
+  }
+  KOKKOS_INLINE_FUNCTION constexpr pointer_type data() const {
+    return static_cast<pointer_type>(base_t::data_handle());
+  }
+
+  KOKKOS_INLINE_FUNCTION constexpr int extent_int(size_t r) const {
+    return static_cast<int>(base_t::extent(r));
+  }
+  //----------------------------------------
+  // Allow specializations to query their specialized map
+
+  KOKKOS_INLINE_FUNCTION
+  auto impl_map() const {
+    using map_type =
+        Kokkos::Impl::ViewMapping<traits, typename traits::specialize>;
+    using offset_type = typename map_type::offset_type;
+    return map_type(
+        data(), offset_type(std::integral_constant<unsigned, 0>(), layout()));
+  }
+
+  KOKKOS_INLINE_FUNCTION
+  const Kokkos::Impl::SharedAllocationTracker& impl_track() const {
+    return base_t::data_handle().tracker();
```

Comment:
Won't this break if the view is unmanaged now?

Reply:
I guess it only fails to compile when you actually try to call it? Its weird. Yeah, but we should just return default constructed SharedAllocationTracker.